### PR TITLE
Add multi-coin TM market execution on Apple MPS

### PR DESCRIPTION
**Pull request by enarjord**
## Summary
- add baseline ordinary market execution to the multi-coin Trailing Martingale Metal proxy for long-only, short-only, fused long+short, hedge, and one-way runs
- retain generation-time market intent through the next valid candle, apply adverse directional slippage/taker fees, executable-touch minimum sizing, and market-touch portfolio entry allocation
- keep recursive entry/close modes, exposure reducers, auto-unstuck, and realized-loss gating fail closed until their dedicated parity slices
- document the supported scope and add runtime/bounds/static-override rejection coverage

Exact Rust remains authoritative; the CPU optimizer and normal bot/backtest paths are unchanged.

## Validation
- `cargo test --no-default-features` (267 passed)
- `cargo check --tests`
- full non-MPS Python suite
- `pytest -q tests/optimization/test_gpu_backend.py`
- full physical Apple M3 MPS suite (401 passed)
- Rust source fingerprint and isolated ABI3 artifact stamp match
- cached Bybit BTC+ETH HSL long-only smoke: 576 proxy / 72 exact
- cached Bybit BTC+ETH HSL fused long+short one-way smoke: 576 proxy / 72 exact
- `python src/tools/check_ai_docs.py`
- `python -m compileall -q src tests/optimization/test_gpu_backend.py tests/optimization/test_gpu_mps.py`
- `git diff --check`

### CODE REVIEW

**Comment by enarjord** [comment]
@codex review

File: src/optimization/backends/gpu_backend.py [hunk 1]
```diff
@@ -906,6 +906,125 @@ def _validate_resume_evidence_budget(
...
+
+
+def _validate_tm_multicoin_market_runtime_scope(
+    config: dict, enabled_sides
+) -> None:
+    """Reject multi-coin TM market features not modeled by this slice."""
+
+    unsupported = []
+    max_loss = float(config.get("live", {}).get("max_realized_loss_pct", 1.0))
+    if not math.isclose(max_loss, 1.0, abs_tol=1.0e-12):
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/backends/gpu_backend.py]: Fixed in 34918214d: multi-coin TM market mode now requires live.max_realized_loss_pct to equal 1.0 exactly, with regressions for nextafter-below, near-below, and above-1 values.

**Comment by enarjord** [comment]
@codex review
